h_emitter.h

```diff
@@ -88,17 +89,44 @@ class HEmitter
     {
         // Prefix is generated, if specified, only in _u.h.
         for (Function* f : edl_->trusted_funcs_)
-            out() << prototype(f, true, gen_t_h_, prefix) + ";"
+        {
+            if (gen_t_h_)
+            {
+                out() << prototype(f, true, gen_t_h_, prefix) + ";"
+                      << "";
+                continue;
+            }
+            /* For the host-side header, use the guard to prevent multiple
+             * definitions. */
+            std::string funstr =
+                (prefix == "") ? f->name_ : prefix + "_" + f->name_;
+            out() << "#ifndef " + upper(funstr) << "#define " + upper(funstr)
```

Comment:
I don't think the guards are needed. It is valid in C to redeclare a function multiple times. The guards don't prevent anything in this case.
On the other hand, say two different edl files have the same function "foo", but foo has different signatures, then without guards, the user will get a compile error, which is what we want.